Mage.Sets/src/mage/cards/n/NobleHeritage.java

```diff
@@ -0,0 +1,160 @@
+import mage.target.common.TargetControlledCreaturePermanent;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * @author Grath
+ */
+public final class NobleHeritage extends CardImpl {
+
+    public NobleHeritage(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{W}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.BACKGROUND);
+
+        // Commander creatures you own have “When this creature enters the battlefield and at the beginning of your upkeep, each player may put two +1/+1 counters on a creature they control. For each opponent who does, you gain protection from that player until your next turn.” (You can’t be targeted, dealt damage, or enchanted by anything controlled by that player.)
+        this.addAbility(new SimpleStaticAbility(new GainAbilityAllEffect(new NobleHeritageTriggeredAbility(), Duration.WhileOnBattlefield,
+                StaticFilters.FILTER_CREATURES_OWNED_COMMANDER
+        )));
+    }
+
+    private NobleHeritage(final NobleHeritage card) {
+        super(card);
+    }
+
+    @Override
+    public NobleHeritage copy() {
+        return new NobleHeritage(this);
+    }
+}
+
+class NobleHeritageTriggeredAbility extends TriggeredAbilityImpl {
+
+    public NobleHeritageTriggeredAbility() {
+        super(Zone.BATTLEFIELD, null, false);
+    }
+
+    public NobleHeritageTriggeredAbility(final NobleHeritageTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public NobleHeritageTriggeredAbility copy() {
+        return new NobleHeritageTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.ENTERS_THE_BATTLEFIELD || event.getType() == GameEvent.EventType.UPKEEP_STEP_PRE;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        this.getEffects().clear();
+        boolean result;
+        if (event.getType() == GameEvent.EventType.ENTERS_THE_BATTLEFIELD) {
+            result = event.getTargetId().equals(this.getSourceId());
+        } else {
+            result = event.getPlayerId().equals(this.getControllerId());
+        }
+        if (result) {
+            Permanent enchantment = game.getPermanentOrLKIBattlefield(getSourceId());
+            if (enchantment != null) {
+                Effect effect = new NobleHeritageEffect();
+                this.getEffects().add(effect);
+            } else {
+                result = false;
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public String getRule() {
+        return "When {this} enters the battlefield or at the beginning of your upkeep, each player may put two +1/+1 counters on a creature they control. For each opponent who does, you gain protection from that player until your next turn.";
```

Comment:
First part of this rule should be "When this creature.." as opposed to "When {this}..", the reason being that it causes a text issue with Noble Heritage's rule text as it considers itself {this}. See below:
![image](https://user-images.githubusercontent.com/26198472/217668895-46086586-2db1-4b34-ab1b-ea3e4e6eaaf6.png)
Additionally, the rule should read "enters the battlefield *and* at the beginning of your upkeep" as opposed to "enters the battlefield *or* at the beginning of your upkeep".

Reply:
Done.